ca/config.go

```diff
@@ -313,7 +332,77 @@ func LoadOrCreateSecurityConfig(ctx context.Context, baseCertDir, token, propose
 		}).Debug("loaded node credentials")
 	}
 
-	return NewSecurityConfig(&rootCA, clientTLSCreds, serverTLSCreds), nil
+	return NewSecurityConfig(&rootCA, krw, clientTLSCreds, serverTLSCreds), nil
+}
+
+// RenewTLSConfigNow gets a new TLS cert and key, and updates the security config if provided.  This is similar to
+// RenewTLSConfig, except while that monitors for expiry, and periodically renews, this renews once and is blocking
+func RenewTLSConfigNow(ctx context.Context, s *SecurityConfig, r remotes.Remotes, updates chan CertificateUpdate) error {
+	ctx = log.WithModule(ctx, "tls")
+	log := log.G(ctx).WithFields(logrus.Fields{
+		"node.id":   s.ClientTLSCreds.NodeID(),
+		"node.role": s.ClientTLSCreds.Role(),
+	})
+
+	// Let's request new certs. Renewals don't require a token.
+	rootCA := s.RootCA()
+	tlsKeyPair, err := rootCA.RequestAndSaveNewCertificates(ctx,
+		s.KeyWriter(),
+		"",
+		r,
+		s.ClientTLSCreds,
+		nil)
+	if err != nil {
+		log.WithError(err).Errorf("failed to renew the certificate")
+		if updates != nil {
+			updates <- CertificateUpdate{Err: err}
+		}
+		return err
+	}
+
+	clientTLSConfig, err := NewClientTLSConfig(tlsKeyPair, rootCA.Pool, CARole)
+	if err != nil {
+		log.WithError(err).Errorf("failed to create a new client config")
+		if updates != nil {
+			updates <- CertificateUpdate{Err: err}
+		}
+	}
+	serverTLSConfig, err := NewServerTLSConfig(tlsKeyPair, rootCA.Pool)
+	if err != nil {
+		log.WithError(err).Errorf("failed to create a new server config")
+		if updates != nil {
+			updates <- CertificateUpdate{Err: err}
+		}
+	}
+
+	err = s.ClientTLSCreds.LoadNewTLSConfig(clientTLSConfig)
+	if err != nil {
+		log.WithError(err).Errorf("failed to update the client credentials")
+		if updates != nil {
+			updates <- CertificateUpdate{Err: err}
+		}
```

Comment:
Since I split this particular function out, I think we can return the error and just ignore it in the auto-renew one, and it will retry.  Those continued attempting to set the TLS config objects on the `rootCA` though, even if the previous one failed - I wasn't sure if that was desired or not.

I'll fix these to return.